model/version_db.go

```diff
@@ -405,3 +407,24 @@ func FindProjectForVersion(versionID string) (string, error) {
 	}
 	return v.Identifier, nil
 }
+
+// AbortVersion sets the abort flag on all tasks associated with the version which are in an
+// abortable state. Also handles child patches.
+func AbortVersion(versionId string, reason task.AbortInfo) error {
```

Comment:
Since this doesn't perform any operation on the version itself I think a better name would be `AbortVersionTasks`

Reply:
Good call